cpp/src/io/parquet/page_enc.cu

```diff
@@ -433,8 +433,9 @@ __global__ void __launch_bounds__(128)
                                   max_RLE_page_size(col_g.num_rep_level_bits(), num_vals));
 
       if (num_rows >= ck_g.num_rows ||
-          (values_in_page > 0 && (page_size + fragment_data_size > this_max_page_size)) ||
-          rows_in_page + frag_g.num_rows > max_page_size_rows) {
+          (values_in_page > 0 && (page_size + fragment_data_size > this_max_page_size ||
```

Comment:
why is `values_in_page` used here instead of `rows_in_page`?

Reply:
History? On the reader side there's the problem of pages with 0 rows but many values, so maybe that's why `values_in_page` was originally used?

Reply:
omg, I forgot about 0 row pages :face_exhaling: 
Need to stare at this some more now